src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -483,6 +546,100 @@ class EntryStorage
         index = found ? toIndex : ArraySize(acl);
     }
 
+public:
+    static const chip::TLV::Tag kTagInUse       = chip::TLV::ContextTag(1);
+    static const chip::TLV::Tag kTagFabricIndex = chip::TLV::ContextTag(2);
+    static const chip::TLV::Tag kTagAuthMode    = chip::TLV::ContextTag(3);
+    static const chip::TLV::Tag kTagPrivilege   = chip::TLV::ContextTag(4);
+    static const chip::TLV::Tag kTagSubjects    = chip::TLV::ContextTag(5);
+    static const chip::TLV::Tag kTagTargets     = chip::TLV::ContextTag(6);
+    static const size_t kStorageBufferSize      = 192;
+
+    CHIP_ERROR Serialize(chip::PersistentStorageDelegate * storage, const char * key)
+    {
+        uint8_t buffer[kStorageBufferSize] = { 0 };
+        chip::TLV::TLVWriter writer;
+        writer.Init(buffer, sizeof(buffer));
+        chip::TLV::TLVType container;
+        ReturnErrorOnFailure(writer.StartContainer(chip::TLV::AnonymousTag, chip::TLV::TLVType::kTLVType_Structure, container));
+
+        ReturnErrorOnFailure(writer.Put(kTagInUse, mInUse));
```

Comment:
We have serializers and de-serializers built into the cluster-object representation of the Acl attributes - why are we duplicating this logic here?

Reply:
OK I had a big chat with Andy where we discussed all the issues (code layers and isolation, reuse, desired properties of serialized data, etc.) and we have a plan to move down the serialization/deserialization code from access-control-cluster.cpp while keeping it usable for both clients (module and cluster).

Reply:
I've talked to Marc and I will see whether the serializers/deserializers for the wire representation can be reused. Theoretically a serialization format specific to the internal data could yield some storage savings, but it's unlikely to be significant in the case of the ACL.